addons/mail/models/ir_translation.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.exceptions import AccessError
+
+
+class IrTranslation(models.Model):
+    _inherit = 'ir.translation'
+
+    def _is_dynamic(self):
+        model, field = self.name.split(',')
+        assert model == 'mail.template'
+
+        engine = getattr(self.env[model]._fields[field], 'render_engine', 'inline_template')
+        if engine in ('qweb', 'qweb_view'):
+            if self.env['mail.render.mixin']._is_dynamic_template_qweb(self.value):
+                return True
+        else:
+            if self.env['mail.render.mixin']._is_dynamic_template_inline_template(self.value):
+                return True
+        return False
+
+    def check(self, mode):
+        """ Check access rights of operation ``mode`` on ``self`` for the
+        current user. Raise an AccessError in case conditions are not met.
+        """
+        super().check(mode)
+
+        # if we don't modify translation of mail.template, we ignore
+        translations_for_mail_template = self.filtered(lambda template: template.name.startswith('mail.template,'))
```

Comment:
we probably need to handle all `mail.render.mixin` models here, and if we delegate the check to the actual model, it will take care of honoring the `_unrestricted_rendering` flag etc.

perf-wise it should be ok, we don't edit those translations all the time